curtains/src/androidTest/java/curtains/test/utilities/TestUtils.kt

```diff
@@ -88,4 +102,52 @@ fun View.onAttachedToWindow(onAttachedToWindow: () -> Unit) {
 
     override fun onViewDetachedFromWindow(v: View) = Unit
   })
+}
+
+fun <T : Activity> launchWaitingForFocus(activityClass: KClass<T>): ActivityScenario<T> {
+  val scenario = ActivityScenario.launch(activityClass.java)
+  var hasFocus = scenario.waitForFocus()
+  if (hasFocus == null) {
+    resolveAnr()
+  }
+  hasFocus = scenario.waitForFocus()
+  if (hasFocus == null) {
+    dumpWindowService()
+  }
+  check(hasFocus) {
+    "expected activity to become focused"
+  }
+  return scenario
+}
+
+private fun <T : Activity> ActivityScenario<T>.waitForFocus(): Boolean? {
+  val activityHasWindowFocus = ArrayBlockingQueue<Boolean>(1)
```

Comment:
The only thing that comes to mind to explain the ANR would be that we're blocking the main thread. Here I don't think it can happen, from reading the code, but an example would be that ArrayBlockingQueue has a capacity of 1 so if we somehow got 2 calls to put() (i.e. here or somewhere else the listener removal somehow didn't work) then the 2nd put would block. But we're also calling poll so we're removing the entry put anyway. So doesn't seem like it, but I just feel like there's got to be something related to blocking ops on the main thread.